### DISCUSSION TITLE
Add doc parameter to APIRouter

### DISCUSSION THREAD

**Discussion by PREPONDERANCE** [fastapi/fastapi#11581] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
router = APIRouter(prefix="stargazer", tags=["overview"], __doc__="")
```

### Description

- I would prefer `APIRouter` adding a parameter `description` as in `FastAPI` class

### Operating System

macOS

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

Python 3.11.3

**Comment by aanchlia** [comment]
I think `APIRouter` doesn't have description parameter, but you can do something like this?
```python
router.description = 'description'
```

**Reply by PREPONDERANCE** [reply]
> I think `APIRouter` doesn't have description parameter, but you can do something like this?
> 
> ```python
> 
> router.description = 'description'
> 
> ```

Indeed APIRouter does not support such parameter which is also why I opened up this discussion.

Back to the code snippet, I actually want the doc string displayed in openAPI docs when I click on those API endpoints inheriting this APIRouter. 

So that's not really what I'm anticipating.

**Comment by mattmess1221** [comment, marked as answer]
It's a bit complex since routers can have multiple tags, and multiple routers and routes can have the same tags.

Tag descriptions are officially documented [here](https://fastapi.tiangolo.com/tutorial/metadata/#metadata-for-tags).

**Reply by PREPONDERANCE** [reply]
Thanks for the explanation! I'll go ahead and find a way that works best for me.
